src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,55 @@
 import React from 'react';
+import { Post } from '../../types/Post';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Post,
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    title,
+    user,
+    body,
+    comments,
+  } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">
+          {title}
+        </h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user
+          && (
+            <UserInfo
+              user={user}
+            />
+          )}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      {comments.length > 0
+        ? (
+          <CommentList
+            comments={comments}
+          />
+        )
+        : (
+          <b data-cy="NoCommentsMessage">No comments yet</b>
+        )}
```

Comment:
single-line is much more concise and totally suitable
```suggestion
        ? <CommentList comments={comments} />
        : <b data-cy="NoCommentsMessage">No comments yet</b>
      }
```